Package.swift

```diff
@@ -11,9 +11,21 @@
 */
 
 import PackageDescription
+import class Foundation.ProcessInfo
+
+// We default to a 10.10 minimum deployment target for clients of libSwiftPM,
+// but allow overriding it when building for a toolchain.
+
+let macOSPlatform: SupportedPlatform
+if let deploymentTarget = ProcessInfo.processInfo.environment["SWIFTPM_MACOS_DEPLOYMENT_TARGET"] {
+    macOSPlatform = .macOS(deploymentTarget)
+} else {
+    macOSPlatform = .macOS(.v10_10)
```

Comment:
Let's just bump to whatever we actually support. We'll need to update sourcekit-lsp. Are you fine with that @benlangmuir?

Reply:
> but the toolchain doesn't actually work on macOS < 10.15 anymore

This is news to me, where was this change discussed?

Reply:
Not sure if it was discussed, bu it's the de facto behaviour today, e.g. `libPackageDescription` in the 5.2 toolchain has a 10.15 minimum deployment target. Since 11.4 only supports 10.15+, it seems like the same should be true for the toolchain as well, right?

Reply:
We talked about this offline and decided to keep the current approach (build with 10.15 in the toolchain, but `libSwiftPM` will have 10.10 minimum deployment target)